executor/executor_linux.cc

```diff
@@ -70,14 +94,38 @@ int main(int argc, char** argv)
 	install_segv_handler();
 	use_temporary_dir();
 
-#if defined(__i386__) || defined(__arm__)
+#if defined(__i386__)
 	// mmap syscall on i386/arm is translated to old_mmap and has different signature.
 	// As a workaround fix it up to mmap2, which has signature that we expect.
 	// pkg/csource has the same hack.
 	for (size_t i = 0; i < sizeof(syscalls) / sizeof(syscalls[0]); i++) {
 		if (syscalls[i].sys_nr == __NR_mmap)
 			syscalls[i].sys_nr = __NR_mmap2;
 	}
+
+#endif
+
+#if defined(__arm__)
+	// syscalls_linux.h, which is auto-generated, appears to be off by 0x900000 (or 9437184).
```

Comment:
Let's move this to syz-extract. Is it's responsibility to generate correct numbers, and one day we will fix it. let's not spread these workarounds throughout the code base. For example, `pkg/csource` also uses syscall numbers, and currently it will use the wrong ones.

Reply:
Changing syz-extract is tricky. I did that, but it has a side-effect. The mmap() system call does not actually exist on Linux for ARM32 -- there is no number for it and it is marked obsolete. The mmap() appears to be implemented in a library using mmap2(). That appears to then cause a problem with executing syz-fuzzer. It appears to be expecting the existing the existence of the mmap() as a system call. I suggest accepting the updated code (it is much more of a localized change now after your suggestion of a revised kOutputAddr). Fixing syz-extract to address the issue may entail substantial work and is best done in a branch.  We can explore that in another pull or another thread.

Reply:
I  meant to leave the mmap part here. Only change `syscalls[i].sys_nr == __NR_mmap` to `strcmp(syscalls[i].name, "mmap") == 0`, which should work for both 386 and arm.

And add this to syz-extract:

```
--- a/sys/syz-extract/linux.go
+++ b/sys/syz-extract/linux.go
@@ -103,5 +103,14 @@ unsigned long phys_base;
 unsigned long __phys_addr(unsigned long addr) { return 0; }
 #endif
 `
-       return extract(info, "gcc", args, addSource, true)
+       res, undeclared, err := extract(info, "gcc", args, addSource, true)
+       if arch.target.Arch == "arm" {
+               // Explain why we are doing this.
+               for name := range res {
+                       if strings.HasPrefix(name, arch.target.SyscallPrefix) {
+                               res[name] -= 0x900000
+                       }
+               }
+       }
+       return res, undeclared, err
 }
```